lib/ProductOpener/Display.pm

```diff
@@ -9607,387 +9566,399 @@ JS
 			$shown = 0;
 		}
 
-		if (($shown) and ($nutriment =~ /^-/)) {
-			$class = 'sub';
-			$prefix = lang("nutrition_data_table_sub") . " ";
-			if ($nutriment =~ /^--/) {
-				$prefix = "&nbsp; " . $prefix;
-			}
-		}
-
-		my $name;
-
-		# display nutrition score only when the country is matching
+		# Show the UK nutrition score only if the country is matching
+		# Always show the FR nutrition score (Nutri-Score)
 
 		if ($nid =~ /^nutrition-score-(.*)$/) {
 			# Always show the FR score and Nutri-Score
 			if (($cc ne $1) and (not ($1 eq 'fr'))) {
 				$shown = 0;
 			}
-			else {
-				$name = '<a href="/nutriscore" title="$product_ref->{nutrition_score_debug}">' . $prefix.$Nutriments{$nid}{$lang} . '</a>';
-			}
 		}
 
-		elsif ((exists $Nutriments{$nid}) and (exists $Nutriments{$nid}{$lang})) {
-			$name = $prefix . $Nutriments{$nid}{$lang};
-
-		}
-		elsif ((exists $Nutriments{$nid}) and (exists $Nutriments{$nid}{en})) {
-			$name = $prefix . $Nutriments{$nid}{en};
+		if ($shown) {
+		
+			# Level of the nutrient: 1 for main nutrients, 2 for sub-nutrients, 3 for sub-sub-nutrients
+			my $level = 1;
+			
+			if ($nutriment =~ /^-/) {
+				$level = 2;
+				if ($nutriment =~ /^--/) {
+					$level = 3;
+				}
+			}
```

Comment:
I do love alternate expressions, but when an alternate expression contains another one, I find it more complex to read and understand.

Reply:
Somehow I never got used to alternate expressions so I never use them. I tend to code in the way I think, which may not be the most efficient or concise.

The way I think here:
By default, it's a main nutrient, unless it is a sub nutrient, or a sub sub nutrient.

Just an habit I guess, but the other way: "is it a sub sub nutrient, or a sub nutrient, otherwise it's a main nutrient." is less natural to me.